sdk/src/resource/resource.cc

```diff
@@ -0,0 +1,69 @@
+#include "opentelemetry/sdk/resource/resource.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/resource/resource_detector.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace resource
+{
+
+const std::string TELEMETRY_SDK_LANGUAGE = "telemetry.sdk.language";
+const std::string TELEMETRY_SDK_NAME     = "telemetry.sdk.name";
+const std::string TELEMETRY_SDK_VERSION  = "telemetry.sdk.version";
+
+Resource::Resource(const ResourceAttributes &attributes) noexcept : attributes_(attributes) {}
+
+Resource Resource::Merge(const Resource &other) noexcept
+{
+  ResourceAttributes merged_resource_attributes(attributes_);
+  for (auto &elem : other.attributes_)
+  {
+    if ((merged_resource_attributes.find(elem.first) == merged_resource_attributes.end()) ||
+        (nostd::holds_alternative<std::string>(attributes_[elem.first]) &&
+         nostd::get<std::string>(attributes_[elem.first]).size() == 0))
+    {
+      merged_resource_attributes[elem.first] = elem.second;
+    }
+  }
+  return Resource(merged_resource_attributes);
```

Comment:
There were some recent spec changes that require this to be revisited: https://github.com/open-telemetry/opentelemetry-specification/pull/1345

Reply:
Thanks for noticing it. It's kind of reverse of original logic. Have incorporated these changes.